drivers/sensor/bmp581/bmp581.c

```diff
@@ -0,0 +1,565 @@
+/*
+ * Copyright (c) 2022 Badgerd Technologies B.V.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "bmp581.h"
+
+#include <math.h>
+
+#include <zephyr/device.h>
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/init.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(bmp581, CONFIG_SENSOR_LOG_LEVEL);
+
+#if DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) == 0
+#warning "BMP581 driver enabled without any devices"
+#endif
+
+static int power_up_check(struct bmp581_data *drv);
+static int get_nvm_status(uint8_t *nvm_status, struct bmp581_data *drv);
+static int get_interrupt_status(uint8_t *int_status, struct bmp581_data *drv);
+static int validate_chip_id(struct bmp581_data *drv);
+static int get_osr_odr_press_config(struct bmp581_osr_odr_press_config *osr_odr_press_cfg,
+					struct bmp581_data *drv);
+static int set_osr_config(const struct sensor_value *osr, enum sensor_channel chan,
+			  struct bmp581_data *drv);
+static int set_odr_config(const struct sensor_value *odr, struct bmp581_data *drv);
+static int soft_reset(struct bmp581_data *drv);
+static int set_iir_config(const struct sensor_value *iir, struct bmp581_data *drv);
+static int get_power_mode(enum bmp5_powermode *powermode, struct bmp581_data *drv);
+static int set_power_mode(enum bmp5_powermode powermode, struct bmp581_data *drv);
+
+int reg_read(uint8_t reg, uint8_t *data, uint16_t length, struct bmp581_data *drv)
+{
+	return i2c_burst_read_dt(&drv->i2c, reg, data, length);
+}
+
+int reg_write(uint8_t reg, const uint8_t *data, uint16_t length, struct bmp581_data *drv)
+{
+	return i2c_burst_write_dt(&drv->i2c, reg, data, length);
+}
```

Comment:
Wrappers like this don't really add anything unless the sensor can support multiple bus interfaces (i.e., I2C and SPI). Please remove

Reply:
The sensor also has an SPI interface. I can remove the wrapper or leave it to someone to add SPI support in the future. Please tell me how you want it.

Reply:
Could you please give me some input about that?

Reply:
The way these wrappers are currently implemented won't work when you add SPI (consider the case if you have two instances of this sensor, one attached to a SPI bus and the other to an I2C bus), so I suggest using `i2c_burst_read_dt` and friends directly for now.

Reply:
I removed the wrappers.